src/Symfony/Component/Encryption/Sodium/SodiumKey.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Encryption\Sodium;
+
+use Symfony\Component\Encryption\Exception\InvalidKeyException;
+use Symfony\Component\Encryption\KeyInterface;
+
+/**
+ * @internal
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+final class SodiumKey implements KeyInterface
+{
+    private ?string $secret = null;
+    private ?string $privateKey = null;
+    private ?string $publicKey = null;
+
+    /**
+     * A keypair can only be created from a public and private key.
+     */
+    private ?string $keypair = null;
+
+    public static function create(string $secret, string $keypair): self
+    {
+        $key = self::fromSecret($secret);
+        $key->keypair = $keypair;
+        $key->publicKey = sodium_crypto_box_publickey($keypair);
+        $key->privateKey = sodium_crypto_box_secretkey($keypair);
+
+        return $key;
+    }
+
+    /**
+     * The secret key length should be 32 bytes, but other sizes are accepted.
+     */
+    public static function fromSecret(string $secret): self
+    {
+        $key = new self();
+        if (\SODIUM_CRYPTO_SECRETBOX_KEYBYTES === \strlen($secret)) {
+            $key->secret = $secret;
+        } else {
+            // Trim the key to a good size
+            $key->secret = substr(sha1($secret), 0, \SODIUM_CRYPTO_SECRETBOX_KEYBYTES);
+        }
+
+        return $key;
+    }
+
+    public static function fromPrivateKey(string $privateKey): self
+    {
+        $key = new self();
+        $key->privateKey = $privateKey;
+
+        return $key;
+    }
+
+    public static function fromPrivateAndPublicKeys(string $privateKey, string $publicKey): self
```

Comment:
not used and class is internal

Reply:
I believe these method will be very useful when we use the encryption component for the "symfony secrets".

Reply:
I have the feeling that having some usages of this new component in core would help moving forward here. 
@Nyholm Maybe could you open a PR doing so based on this PR? Or commit here directly?